core/src/main/java/org/elasticsearch/index/query/FilteredQueryParser.java

```diff
@@ -83,44 +78,15 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
             }
         }
 
-        // parsed internally, but returned null during parsing...
-        if (query == null) {
-            return null;
-        }
```

Comment:
this null check seems useful, if no query was found, return null meaning ignore this filtered query as a whole. weird though.